internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -301,14 +301,19 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
 			sylog.Debugf("Received signal %s", s.String())
 			switch s {
 			case syscall.SIGCHLD:
+			InnerLoop:
 				for {
 					var status syscall.WaitStatus
 
 					wpid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, nil)
-					if wpid <= 0 || err != nil {
-						break
-					}
```

Comment:
Would linter be happy with `} else if wpid == cmd.Process.Pid && err == nil {` instead ?